src/ui/React/Documentation.tsx

```diff
@@ -0,0 +1,71 @@
+import React, { useContext, useState } from "react";
+
+interface Navigator {
+  navigate: (s: string, external: boolean) => void;
+}
+
+export const Navigator = React.createContext<Navigator>({ navigate: () => undefined });
+
+export const useNavigator = (): Navigator => useContext(Navigator);
+
+interface History {
+  pages: string[];
+  page: string;
+  push(p: string): void;
+  pop(): void;
+  home(): void;
+}
+
+const defaultPage = "index.md";
+
+const HistoryContext = React.createContext<History>({
+  page: "",
```

Comment:
Use stricter types for History
```suggestion
interface History {
  pages: FilePath[];
  page: FilePath;
  push(p: FilePath): void;
  pop(): void;
  home(): void;
}

const defaultPage = asFilePath("index.md");

const HistoryContext = React.createContext<History>({
  page: defaultPage,
```